### ISSUE TITLE
[Gemma-3] RMSNorm: higher-rank q_norm/k_norm skip the fused CUDA kernel, and mixed-dtype weights return NaNs

### ISSUE THREAD

**Issue by gujialiang123** [sgl-project/sglang#32807]
## Motivation

`Gemma3RMSNorm.forward_cuda` dispatches to the fused kernel only for 2-D inputs. `q_norm` and `k_norm` are called with `[tokens, heads, head_dim]`, so they take the eager path instead — **52 of the 157 norm calls per forward** on `gemma-3-1b`.

Separately, both fused paths pass `self.weight.data` without checking its dtype, and the fused kernels return NaNs rather than raising on a mismatch.

### 1. Higher-rank inputs are not fused

On current `main` (`python/sglang/srt/layers/layernorm.py`):

```python
def forward_cuda(self, x, residual: Optional[torch.Tensor] = None):
    if residual is not None:
        gemma_fused_add_rmsnorm(x, residual, self.weight.data, self.eps)
        return x, residual
    if x.dim() == 2:
        return gemma_rmsnorm(x, self.weight.data, self.eps)
    return self.forward_native(x)      # <- q_norm / k_norm land here
```

#21962 proposed a general high-rank reshape; #32383 later landed the 2-D and residual paths. The `x.dim() == 2` guard is what remains.

The eager path is `pow -> mean -> add -> rsqrt -> mul` plus fp32 up/down casts — roughly six kernels and seven HBM round trips per norm, where one fused kernel would do. RMSNorm reduces over the last dimension only, so flattening the leading dimensions and restoring the shape is exact.

### 2. Mixed-dtype weight returns NaNs instead of raising

`self.weight` is `nn.Parameter(torch.zeros(dim))`, so its dtype follows whatever default is in effect at construction. Under the loader's `set_default_torch_dtype` it matches the activations, so **this is a robustness gap rather than a live bug on `main`**. But a module built outside that context keeps an fp32 weight, and then:

```python
gemma_rmsnorm(bf16_input, bf16_weight)  # -> finite
gemma_rmsnorm(bf16_input, fp32_weight)  # -> NaN/Inf, no exception
```

## Reproduction

Both observations are reproducible without a server:

```python
import torch
from sglang.srt.layers.layernorm import Gemma3RMSNorm

torch.set_default_device("cuda")

# (1) higher-rank input silently takes the eager path
n = Gemma3RMSNorm(256).cuda()
n.weight.data.normal_(std=0.1)
n.weight.data = n.weight.data.to(torch.bfloat16)
x3 = (torch.randn(37, 4, 256) * 0.02).to(torch.bfloat16)
print("3-D output identical to forward_native:",
      torch.equal(n.forward_cuda(x3), n.forward_native(x3)))   # True => not fused

# (2) fp32 weight against bf16 activations returns NaNs
from sgl_kernel import gemma_rmsnorm
x = (torch.randn(64, 1152) * 0.02).to(torch.bfloat16).contiguous()
for name, w in (("bf16", (torch.randn(1152) * 0.1).to(torch.bfloat16)),
                ("fp32", torch.randn(1152) * 0.1)):
    out = gemma_rmsnorm(x, w.contiguous(), 1e-6)
    print(f"{name} weight -> NaN/Inf:", bool(torch.isnan(out).any() or torch.isinf(out).any()))
```

## Impact

Measured as the increment attributable to fusing the high-rank path only, i.e. against a build whose norm coverage matches current `main` (2-D fused, higher rank eager). `gemma-3-1b-it`, 1×H200, BF16, 6 reps per arm, Welch t-test:

| regime | baseline | patched | gain | p |
|---|---|---|---|---|
| low-batch decode | 1.300 req/s | 1.776 req/s | **+36.6%** | 2.4e-14 |
| concurrent decode | 33.648 req/s | 41.891 req/s | **+24.5%** | 1.2e-06 |
| 4K prefill-heavy (in=4000, out=32, conc=4) | 23.385 req/s | 25.085 req/s | +7.3% | 0.053 (not significant) |

## Environment

Captured during the experiments (all 8 GPUs on this host are currently occupied by another user, so `sglang.check_env` could not be run just now without allocating a context):

```
GPU              : NVIDIA H200
Driver           : 580.105.08
torch            : 2.9.1+cu128
triton           : 3.5.1
sglang           : 0.5.12.post1
CUDA             : 12.8
dtype            : BF16, TP1
model            : google/gemma-3-1b-it
```

The code observations above were re-verified against `main` at `4e5a05148a`.

## Proposed fix

Dispatch higher-rank inputs through the fused kernel by flattening the leading dimensions, and guard the fused paths so a dtype/device mismatch falls back to `forward_native` rather than producing NaNs. Falling back (instead of casting) preserves fp32-weight semantics, keeps the residual path byte-identical to `main`, and adds no state to the module.

PR: #32670
